src/components/interface/AnalysisFooter.tsx

```diff
@@ -82,17 +90,39 @@ export function AnalysisFooter() {
               }}
               data={allParticipants || []}
             />
-            <Button onClick={() => navigate(`../${funcIndex ? '..' : ''}${encryptIndex(+currentStep - 1)}?participantId=${participantId}`, { relative: 'path' })}>
+            <Button
+              disabled={isStart}
+              onClick={() => {
+                if (funcIndex) {
+                  if (decryptIndex(funcIndex) > 0) {
+                    navigate(`../${encryptIndex(decryptIndex(funcIndex) - 1)}?participantId=${participantId}`, { relative: 'path' });
+                  } else {
+                    navigate(`../../${encryptIndex(+currentStep - 1)}?participantId=${participantId}`, { relative: 'path' });
+                  }
+                } else {
+                  navigate(`../${encryptIndex(+currentStep - 1)}?participantId=${participantId}`, { relative: 'path' });
+                }
+              }}
+            >
               <IconArrowLeft />
             </Button>
-            <Button onClick={() => navigate(`../${funcIndex ? '..' : ''}${encryptIndex(+currentStep + 1)}?participantId=${participantId}`, { relative: 'path' })}>
+            <Button
+              disabled={isEnd}
+              onClick={() => {
+                if (funcIndex) {
+                  navigate(`../${encryptIndex(decryptIndex(funcIndex) + 1)}?participantId=${participantId}`, { relative: 'path' });
+                } else {
```

Comment:
What if we're at the end of the dynamic block? Think this needs a bit more caution/another check for that